devops/scripts/benchmarks/utils/unitrace.py

```diff
@@ -0,0 +1,182 @@
+        return m.group(1), m.group(2)
+    return None, None
+
+
+def prune_unitrace_dirs(base_dir, FILECNT=10):
+    """
+    Keeps only FILECNT newest directories for each save_name group in base_dir.
+    """
+    dirs = [d for d in os.listdir(base_dir) if os.path.isdir(os.path.join(base_dir, d))]
+    groups = {}
+    for d in dirs:
+        save_name, ts = extract_save_name_and_timestamp(d)
+        if save_name and ts:
+            groups.setdefault(save_name, []).append((d, ts))
+    for save_name, dirlist in groups.items():
+        # Sort by timestamp string (lexicographically, works for YYYYMMDD_HHMMSS)
+        dirlist.sort(key=lambda x: x[1])
+        if len(dirlist) > FILECNT:
+            for d, ts in dirlist[: len(dirlist) - FILECNT]:
+                full_path = os.path.join(base_dir, d)
+                print(f"Removing old unitrace dir: {full_path}")
+                shutil.rmtree(full_path)
+
+
+def unitrace_cleanup(bench_cwd, unitrace_output):
+    # Remove .pid files from the benchmark directory and .json files from cwd
+    unitrace_dir = os.path.dirname(unitrace_output)
+    unitrace_base = os.path.basename(unitrace_output)
+    print(f"Cleanup unitrace output {unitrace_base} from {unitrace_dir}")
+    for f in os.listdir(unitrace_dir):
+        if f.startswith(unitrace_base + "."):
+            os.remove(os.path.join(unitrace_dir, f))
+            print(f"Cleanup: Removed {f} from {unitrace_dir}")
+    if os.path.exists(bench_cwd):
+        for f in os.listdir(bench_cwd):
+            if f.endswith(".json"):
+                os.remove(os.path.join(bench_cwd, f))
+                print(f"Cleanup: Removed {f} from {bench_cwd}")
+
+
+def unitrace_prepare(name, unitrace_timestamp, command, extra_unitrace_opt=[]):
+    unitrace_bin = os.path.join(options.workdir, "unitrace-build", "unitrace")
+    if not os.path.exists(unitrace_bin):
+        raise FileNotFoundError(f"Unitrace binary not found: {unitrace_bin}. ")
+    os.makedirs(options.unitrace_res_dir, exist_ok=True)
+    if not options.save_name:
+        raise ValueError(
+            "Unitrace requires a save name to be specified via --save option."
+        )
+    bench_dir = f"{options.unitrace_res_dir}/{options.save_name}_{unitrace_timestamp}"
+    os.makedirs(bench_dir, exist_ok=True)
+
+    unitrace_output = f"{bench_dir}/{name}_{unitrace_timestamp}"
+    unitrace_command = (
+        [
+            str(unitrace_bin),
+            "--call-logging",
+            "--host-timing",
+            "--device-timing",
+            "--chrome-sycl-logging",
+            "--chrome-call-logging",
+            "--chrome-kernel-logging",
+            "--output",
+            unitrace_output,
+        ]
+        + extra_unitrace_opt
+        + command
+    )
+    if options.verbose:
+        print(f"Unitrace cmd: {' '.join(unitrace_command)}")
+
+    return bench_dir, unitrace_output, unitrace_command
+
+
+def handle_unitrace_output(bench_dir, unitrace_output, timestamp):
```

Comment:
I feel like this is all very complicated. Maybe we can use directory structure to help us?

Let's say we have two benchmarks: "SubmitKernel" and "Hashtable". We can organize the unitrace output such that each trace file is in its own directory:
```
.../results/traces/SubmitKernel/YYYYMMDD_HHMMSS_[name].PID.out
.../results/traces/Hashtable/YYYYMMDD_HHMMSS_[name].PID.out
```

Removing old results would become simply removing all but 5 newest files in a directory. We wouldn't have to find, rename and move files, since we could make unitrace output them to the destination place upfront.